reproject/mosaicking/tests/test_wcs_helpers.py

```diff
@@ -191,8 +173,62 @@ def test_invalid_wcs_shape(self):
 
     def test_invalid_not_celestial(self):
 
-        self.wcs.wcs.ctype = "OFFSETX", "OFFSETY"
+        self.wcs = self.generate_wcs(celestial=False)
 
         with pytest.raises(TypeError) as exc:
             wcs, shape = find_optimal_celestial_wcs([(self.array, self.wcs)])
         assert exc.value.args[0] == "WCS does not have celestial components"
+
+
+class TestOptimalFITSWCS(BaseTestOptimalWCS):
+    def generate_wcs(
+        self, crpix=(10, 15), crval=(43, 23), cdelt=(-0.1, 0.1), pc=None, celestial=True
+    ):
+        wcs = WCS(naxis=2)
+        if celestial:
+            wcs.wcs.ctype = "RA---TAN", "DEC--TAN"
+        else:
+            wcs.wcs.ctype = "OFFSETX", "OFFSETY"
+        wcs.wcs.crpix = crpix
+        wcs.wcs.crval = crval
+        wcs.wcs.cdelt = cdelt
+        wcs.wcs.equinox = 2000.0
+        if pc is not None:
+            wcs.wcs.pc = pc
+        return wcs
+
+    crval_atol = 1e-8
+    crpix_atol = 1e-6
+    cdelt_rtol = 1e-8
+
+    identity_expected_crpix = 10, 15
+    auto_rotate_expected_crpix = 10, 15
+    multiple_size_expected_crpix = 20, 15
+
+    # The following values are empirical and just to make sure there are no regressions
+    frame_projection_expected_crpix = 16.212189, 28.861195
+    frame_projection_expected_shape = 47, 50
+
+
+class TestOptimalAPE14WCS(TestOptimalFITSWCS):
```

Comment:
Is it more correct like this?
```suggestion
class TestOptimalAPE14WCS(BaseTestOptimalWCS):
```